### PR TITLE
helm3: Enabling upgrading control planes for testing

### PR DESCRIPTION
**Pull request by rossf7**
Some final things found while testing the control plane upgrades.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,8 +32,6 @@ workflows:
           requires:
             - go-build
           filters:
-            branches:
-              only: master
```
Comment (rossf7) [inline .circleci/config.yml:36]: Hit this testing on giraffe. We should always push to aliyun to make testing easier.

File: service/controller/app/resource/releasemigration/create.go [hunk 2]
```diff
@@ -38,17 +38,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
-	//TODO: Remove this statement when we need to test on control planes.
-	if key.InCluster(cr) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q uses InCluster kubeconfig no need to migrate releases", key.AppName(cr)))
-		r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling the resource")
-		return nil
-	}
```
Comment (rossf7) [inline service/controller/app/resource/releasemigration/create.go:46]: Removing this as I don't think we need it anymore.

File: service/controller/app/resource/releasemigration/create.go [hunk 3]
```diff
@@ -38,17 +38,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
-
 	// Resource is used to migrating Helm 2 release into Helm 3 in case of chart-operator app reconciliation.
 	// So for other apps we can skip this step.
 	if key.AppName(cr) != key.ChartOperatorAppName {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to migrating Helm release for %#q", key.AppName(cr)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to migrate release for %#q", key.AppName(cr)))
```
Comment (rossf7) [inline service/controller/app/resource/releasemigration/create.go:44]: Small edit for readability.

File: service/controller/app/resource/releasemigration/resource.go [hunk 4]
```diff
@@ -29,6 +29,7 @@ type Config struct {
 
 	// Settings.
 	ChartNamespace string
+	ImageRegistry  string
```
Comment (rossf7) [inline service/controller/app/resource/releasemigration/resource.go:32]: We should set the registry so we use Aliyun for the migration chart in China.

File: service/controller/app/resource/releasemigration/resource.go [hunk 5]
```diff
@@ -162,17 +171,20 @@ func (r *Resource) ensureReleasesMigrated(ctx context.Context, k8sClient k8sclie
 			return microerror.Mask(err)
 		}
 		if len(releases) > 0 {
-			return microerror.Maskf(executionFailedError, "helm v2 releases not deleted: %#q", releases)
+			desc := fmt.Sprintf("%d helm v2 releases not migrated", len(releases))
+			r.logger.Log("level", "debug", "message", desc)
+
+			return microerror.Maskf(executionFailedError, desc)
```
Comment (rossf7) [inline service/controller/app/resource/releasemigration/resource.go:177]: Extending the logging so you can see the migration progress.

File: service/controller/app/resource/releasemigration/resource.go [hunk 6]
```diff
@@ -162,17 +171,20 @@ func (r *Resource) ensureReleasesMigrated(ctx context.Context, k8sClient k8sclie
...
 		return nil
 	}
 
 	n := func(err error, t time.Duration) {
-		r.logger.Log("level", "debug", "message", "failed to delete all helm v2 releases")
+		r.logger.Log("level", "debug", "message", "failed to migrate all helm v2 releases")
 	}
 
-	b := backoff.NewConstant(backoff.ShortMaxWait, backoff.ShortMaxInterval)
+	b := backoff.NewConstant(5*time.Minute, 10*time.Second)
```
Comment (rossf7) [inline service/controller/app/resource/releasemigration/resource.go:187]: 2 mins was too short for control planes. But in giraffe completed in under 5 mins.

**Comment by rossf7** [comment]
Tested on giraffe.

File: service/controller/app/resource/releasemigration/create.go (hunk 2 above)
Comment (tomahawk28) [reply service/controller/app/resource/releasemigration/create.go:46]: yeah, it only reconciled on chart-operator app CR.

**Pull request merged**